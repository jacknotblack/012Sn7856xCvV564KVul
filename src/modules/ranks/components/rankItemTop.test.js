import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RankItemTop from "./rankItemTop";

describe("A suite", () => {
  let wrapper;
  const props = {
    rankNumber: 2,
    name: "Iron Man",
    stars: 95167456314675,
    bg: "../assets/2.3.1.1_ranking_gastby_daily/bg_second_ranking.svg"
  };

  beforeEach(() => {
    wrapper = shallow(<RankItemTop {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render name", () => {
    expect(wrapper.find(".name").text()).toEqual(props.name);
  });

  it("should render star numbers", () => {
    expect(wrapper.find(".star-wrapper").text()).toEqual(
      props.stars.toString()
    );
  });

  it("should render rank numbers", () => {
    expect(wrapper.find(".rank-number").text()).toEqual(
      `No. ${props.rankNumber}`
    );
  });

  it("should add class according to rank number to rank-number div", () => {
    expect(wrapper.find(".rank-number").hasClass(`no${props.rankNumber}`));
  });

  it("should render right avatar wrapper", () => {
    expect(
      wrapper
        .find(".avatar-wrapper img")
        .at(1)
        .prop("src")
    ).toEqual(props.bg);
  });
});
