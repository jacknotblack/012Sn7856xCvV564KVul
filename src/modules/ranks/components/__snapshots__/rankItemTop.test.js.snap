// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should match snapshot 1`] = `
<div
  className="rank-item-top no2"
>
  <div
    className="rank-number no2"
  >
    No. 
    2
  </div>
  <div
    className="avatar-container"
  >
    <div
      className="avatar-wrapper"
    >
      <img
        alt="avatar"
        src="avatar.jpg"
      />
      <img
        alt="firstPlaceBG"
        src="../assets/2.3.1.1_ranking_gastby_daily/bg_second_ranking.svg"
      />
    </div>
    <span
      className="name"
    >
      Iron Man
    </span>
    <div
      className="star-wrapper"
    >
      95167456314675
    </div>
  </div>
</div>
`;
